source/violation.c

```diff
@@ -399,3 +407,189 @@ candl_violation_p candl_violation(osl_scop_p orig_scop,
   return violation;
 }
 
+static candl_violation_p violation_helper(osl_dependence_p orig_dependence,
+    osl_relation_p context, osl_statement_p source, osl_statement_p target,
+    int nb_parameters, candl_options_p options, PipOptions * pip_options) {
```

Comment:
Ok, do you judge the best.
From my point of view, the documentation should not juste be for publicly exposed functions ans not for small or static functions, because it only cares for the library users, and not maintainers/developers.
If a new dev is added to the project, and you're not there anymore to give him some guidance, some non-documented code you wrote can be incomprehensible to him.
But like I said, do what you judge to be the best.